woocommerce/woocommerce-gutenberg-products-block
# Fix product grid blocks markup & design inconsistencies

Part of #1916.
Fixes #2107.
Fixes #2231.
Fixes #2414.

Changes included in this PR:
- Images in product grid blocks now occupy all the available width.
- _All Products_ prices have the same style as the 'legacy' product grid blocks.
- `Sale!` has been renamed to `Sale` and now it's a `div` instead of a `span` (shouldn't affect layout).
- There are some other small CSS changes.

### Screenshots

_Before (Handpicked products on top, All Products below):_
![imatge](https://user-images.githubusercontent.com/3616980/83164436-828a4280-a10c-11ea-81c1-b9a62cdf52b5.png)


_After (Handpicked products on top, All Products below):_
![imatge](https://user-images.githubusercontent.com/3616980/83166453-3d1b4480-a10f-11ea-813f-2515b26dedac.png)


cc @LevinMedia

### How to test the changes in this Pull Request:

1. Create a page with the _Handpicked products_ block and the _All Products_ block.
2. Both, in the editor and the frontend, verify they look almost the same. There might still be some differences like the _Sale_ badge position, but that can be changed by the user with the inner blocks. Product name colors are different per https://github.com/woocommerce/storefront/issues/1366.
3. Test the CSS snippets in the release notes section below and make sure they undo the changes introduced in this PR.

### Release notes
Images in product grid blocks now expand to occupy all the available horizontal space if they are small. You can undo this change with this CSS snippet:
```CSS
.wc-block-grid__products .wc-block-grid__product-image img {
	width: auto;
}
```

Prices in the _All Products_ block follow the same layout as the other product grid blocks. You can undo this change with:
```CSS
.wc-block-grid__product-price .wc-block-grid__product-price__regular {
	font-size: 0.8em;
	line-height: 1;
	color: #555;
	margin-top: -0.25em;
	display: block;
}
.wc-block-grid__product-price .wc-block-grid__product-price__value {
	letter-spacing: -1px;
	font-weight: 600;
	display: block;
	font-size: 1.25em;
	line-height: 1.25;
	color: #000;
	margin-left: 0;
}
.wc-block-grid__product-price .wc-block-grid__product-price__value span {
	white-space: nowrap;
}
```

### Changelog

> The All Products block and the other product grid blocks now share more styles and the markup is more similar (see release post or docs to learn how to undo this change).
